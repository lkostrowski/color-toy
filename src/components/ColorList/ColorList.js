import React from 'react';
import PropTypes from 'prop-types';
import {IColor} from "../../models/Color.interface";
import {ColorLabel, ColorThumb, Item, List} from "./ColorList.ui";

/**
 *Colors list with color thumbnail and label
 */
const ColorList = ({items, selectedColor = {}, onColorClicked, ...p}) => (
    <List {...p}>
        {items.map(color => (
            <Item onClick={() => onColorClicked(color)}
                  key={color.name}
                  color={color.hex}
                  active={color.name === selectedColor.name}
            >
                <ColorThumb color={color.hex}/>
                <ColorLabel>{color.name}</ColorLabel>
            </Item>
        ))}
    </List>
);

ColorList.propTypes = {
    items: PropTypes.arrayOf(IColor),
    selectedColor: IColor,
    onColorClicked: PropTypes.func.isRequired
};

export default ColorList;
