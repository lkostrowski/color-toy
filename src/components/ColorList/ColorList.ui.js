import glamorous from 'glamorous';

export const List = glamorous.ul({
    margin: 0,
    padding: 0,
    listStyle: 'none',
});

export const Item = glamorous.li(
    {
        display: 'flex',
        alignItems: 'center',
        height: '2rem',
        padding: '1rem',
        margin: 0,
        border: '2px solid transparent',
        cursor: 'pointer',
    },
    p => ({
        borderColor: p.active && p.color,
    }),
);

export const ColorThumb = glamorous.div(
    {
        width: '1rem',
        height: '1rem',
        marginRight: '1rem',
    },
    p => ({
        background: p.color,
    }),
);

export const ColorLabel = glamorous.span({});
