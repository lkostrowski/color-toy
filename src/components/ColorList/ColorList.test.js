import ColorList from './ColorList';
import React from 'react';
import renderer from 'react-test-renderer';

describe('ColorList component', () => {
    it('render expected html', () => {
        const noop = () => {
        };
        const testColors = [
            {
                name: 'test-color1',
                hex: '#ffffff',
            },
            {
                name: 'test-color2',
                hex: '#000000',
            },
        ];
        const testActiveColor = testColors[0];


        const markup = renderer.create(<ColorList
            onColorClicked={noop}
            selectedColor={testActiveColor}
            items={testColors}
        />).toJSON();

        expect(markup).toMatchSnapshot();
    });
});
