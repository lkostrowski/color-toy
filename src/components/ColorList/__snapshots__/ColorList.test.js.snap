// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorList component render expected html 1`] = `
<ul
  className="css-e03p5w"
>
  <li
    className="css-1oxu79"
    onClick={[Function]}
  >
    <div
      className="css-x2kxff"
    />
    <span
      className="css-nil"
    >
      test-color1
    </span>
  </li>
  <li
    className="css-1x7obq3"
    onClick={[Function]}
  >
    <div
      className="css-gbdd7j"
    />
    <span
      className="css-nil"
    >
      test-color2
    </span>
  </li>
</ul>
`;
